src/link.js

```diff
@@ -75,17 +61,69 @@ module.exports = function link(config, args) {
     asset => path.basename(asset)
   );
 
-  if (isEmpty(assets)) {
-    return;
-  }
+  const tasks = dependencies.map((dependency) => (next) => {
```

Comment:
I thought that hooks were going to happen transparently for the plugin in the `rnpm` instead of `rnpm-link`. That will require quite a lot of work to be done by the community authors. I generally like the approach but would prefer it to run in rnpm cli.